ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -271,8 +272,9 @@ private boolean shouldReplayEvent(FileStatus dir, DumpType dumpType) throws Sema
       try {
         tbl = Hive.get().getTable(dbNameOrPattern, tblNameOrPattern);
       } catch (HiveException e) {
-        LOG.error("failed to get the table " + dbNameOrPattern + "." + tblNameOrPattern);
-        throw new SemanticException(e);
+        // may be the table is getting created in this load
+        LOG.info("failed to get the table " + dbNameOrPattern + "." + tblNameOrPattern);
+        return true;
       }
       if (tbl != null) {
         Map<String, String> params = tbl.getParameters();
```

Comment:
This code snippet to validate repl state is duplicated for database case as well. Can we have a static method in ReplicationSpec which takes current event id and parameters map as arguments?

Reply:
done